topic-operator/src/main/java/io/strimzi/operator/topic/KafkaStreamsConfiguration.java

```diff
@@ -0,0 +1,219 @@
+                                storeName,
+                                Serdes.String(), new TopicSerde()
+                        ),
+                filterPredicate
+        );
+    }
+
+    private AsyncBiFunctionServiceGrpc.AsyncBiFunctionServiceImplBase streamsAsyncBiFunctionServiceGrpcImpl(
+            LocalService<AsyncBiFunctionService.WithSerdes<String, String, Integer>> localWaitForResultService
+    ) {
+        return new AsyncBiFunctionServiceGrpcLocalDispatcher(Collections.singletonList(localWaitForResultService));
+    }
+
+    private Lifecycle streamsGrpcServer(
+            HostInfo localHost,
+            KeyValueStoreGrpc.KeyValueStoreImplBase streamsStoreGrpcImpl,
+            AsyncBiFunctionServiceGrpc.AsyncBiFunctionServiceImplBase streamsAsyncBiFunctionServiceGrpcImpl
+    ) {
+        UnknownStatusDescriptionInterceptor unknownStatusDescriptionInterceptor =
+                new UnknownStatusDescriptionInterceptor(
+                        Map.of(
+                                IllegalArgumentException.class, Status.INVALID_ARGUMENT,
+                                IllegalStateException.class, Status.FAILED_PRECONDITION,
+                                InvalidStateStoreException.class, Status.FAILED_PRECONDITION,
+                                Throwable.class, Status.INTERNAL
+                        )
+                );
+
+        Server server = ServerBuilder
+                .forPort(localHost.port())
+                .addService(
+                        ServerInterceptors.intercept(
+                                streamsStoreGrpcImpl,
+                                unknownStatusDescriptionInterceptor
+                        )
+                )
+                .addService(
+                        ServerInterceptors.intercept(
+                                streamsAsyncBiFunctionServiceGrpcImpl,
+                                unknownStatusDescriptionInterceptor
+                        )
+                )
+                .build();
+
+        return new Lifecycle() {
+            @Override
+            public void start() {
+                try {
+                    server.start();
+                } catch (IOException e) {
+                    throw new UncheckedIOException(e);
+                }
+            }
+
+            @Override
+            public void stop() {
+                ConcurrentUtil
+                        .<Server>consumer(Server::awaitTermination)
+                        .accept(server.shutdown());
+            }
+
+            @Override
+            public boolean isRunning() {
+                return !(server.isShutdown() || server.isTerminated());
+            }
+        };
+    }
+
+    public void stop() {
+        Collections.reverse(closeables);
+        closeables.forEach(KafkaStreamsConfiguration::close);
+    }
+
+    public TopicStore getTopicStore() {
+        return topicStore;
+    }
+
+    private static void close(Object service) {
```

Comment:
Why do we not just change this to only take Objects of type `AutoCloseable` ?

Reply:
I guess this was a copy/paste, where some proxies got auto-generated/added AutoClosable, but was not part of the interface itself.
Let me check if this change would work here -- if all references are actually AutoCloseable.